### DISCUSSION TITLE
How to load content into l-popup dynamically (on open)?

### DISCUSSION THREAD

**Discussion by 1Luc1** [vue-leaflet/vue-leaflet#315] [Q&A] [answered]
I'm struggling loading content into l-popup when the popup is opening. Right now I'm using `@popupopen` of `l-marker` to set a temporary variable `popupRoom` which will be rendered as the popup content. First time, it does work, unfortunately, after changing the markers (`rooms` list is changing cause of filtering) and clicking a marker again, the `popupRoom` variable shows a wrong value.

```html
<l-marker
    :icon="circleIcon"
    v-for="(room, index) in rooms"
    :key="'marker-' + index"
    :lat-lng="latLng2(room)"
    @popupopen="popupopen(room.id)"
    @popupclose="popupclose"
  >
  <l-popup>
      <v-card flat>
        ID: {{ room.id }}
        <br />
        PopupRoom: {{ popupRoom?.id }}
      </v-card>
    </l-popup>
</l-marker>
```

```js
const rooms = ref([]);
const popupRoom = ref({});

...

function popupopen(roomId) {
  // get room content from api
  popupRoom.value.id = roomId;
}
```

First time loading rooms:
![grafik](https://github.com/vue-leaflet/vue-leaflet/assets/25635952/8d665eaa-9ae2-401e-b8fa-70dddd81ddc5)

After rooms are reloaded / filtered:
![grafik](https://github.com/vue-leaflet/vue-leaflet/assets/25635952/2bbacc78-22ac-4a02-bdd8-275a25f6598f)

**Comment by veitbjarsch** [comment]
Might be because you use a ref instead of a reactive. If you use a ref you need to update the whole value to achieve proper reactivity.

**Reply by 1Luc1** [reply, marked as answer]
Thanks for your reply. The showed example was kind of a simplification of my actual usage / problem. I tried your suggestions without any changes. Nevertheless, looking into it again, it seems this  call  `@popupopen="popupopen(room.id)"` didn't use the correct `room.id` . Changing it to the following code works for now:

```html
<l-marker
  ...
 @popupopen="popupopen(index)"
  ...
>
  ...
</l-marker>
```

```js
function popupopen(index) {
  // get room content from api
  popupRoom.value.id = rooms.value[index].id;
}
```
